modules/nf-core/nonpareil/tests/main.nf.test

```diff
@@ -0,0 +1,87 @@
+nextflow_process {
+
+    name "Test Process NONPAREIL"
+    script "../main.nf"
+    process "NONPAREIL"
+    config "./nextflow.config"
+    tag "modules"
+    tag "modules_nfcore"
+    tag "nonpareil"
+
+    test("candidatus_portiera_aleyrodidarum - gzipped") {
+
+        when {
+            params {
+                // define parameters here. Example:
+            // outdir = "tests/results"
+            }
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:false ], // meta map
+                    file(params.test_data['candidatus_portiera_aleyrodidarum']['illumina']['test_1_fastq_gz'], checkIfExists: true)
+                ]
+                input[1] = 'fastq'
+                input[2] = 'kmer'
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert process.out.npa.get(0).get(1) ==~ ".*/test.npa" },
+                { assert process.out.npc.get(0).get(1) ==~ ".*/test.npc" },
+                { assert path(process.out.npo.get(0).get(1)).getText().contains("# @impl: Nonpareil") },
+                { assert snapshot(path(process.out.npl.get(0).get(1))).match() },
```

Comment:
```suggestion
                { assert snapshot(
                    file(process.out.npa[0][1]).name,
                    file(process.out.npc[0][1]).name,
                    path(process.out.npo[0][1]).readLines()[0],
                    path(process.out.npl[0][1]
                    ).match()
                }
                    
```

I think something like this might be easier to maintain :) This will snapshot the filename (without the full path) of the first two files, the first line of the 3rd file (please change this to whatever line you want to check since I don't know this format) and the md5sum of the last file

Reply:
Ooooh much more readable!